src/main/java/http/HttpResponse.java

```diff
@@ -0,0 +1,81 @@
+package http;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+public class HttpResponse {
+	private static final String HEADER_CONTENT_LENGTH = "Content-Length";
+	private static final String HEADER_CONTENT_TYPE = "Content-Type";
+
+	private final HttpStatus httpStatus;
+	private final HttpHeader httpHeader;
+	private final HttpResponseBody httpResponseBody;
+
+	public HttpResponse(HttpStatus httpStatus, HttpHeader httpHeader, HttpResponseBody httpResponseBody) {
+		this.httpStatus = httpStatus;
+		this.httpHeader = httpHeader;
+		this.httpResponseBody = httpResponseBody;
+	}
+
+	public static HttpResponse movedPermanently(String redirectURI) {
+		return new HttpResponse(HttpStatus.MOVED_PERMANENTLY, new HttpHeader(Map.of("Location", redirectURI)),
+			new HttpResponseBody());
+	}
+
+	public static HttpResponse found(String redirectURI) {
+		return new HttpResponse(HttpStatus.FOUND, new HttpHeader(Map.of("Location", redirectURI)),
+			new HttpResponseBody());
+	}
+
+	public static HttpResponse badRequest(String errorMessage) {
+		return new HttpResponse(HttpStatus.BAD_REQUEST, new HttpHeader(), new HttpResponseBody(errorMessage));
+	}
+
+	public static HttpResponse notFound(String errorMessage) {
+		return new HttpResponse(HttpStatus.NOT_FOUND, new HttpHeader(), new HttpResponseBody(errorMessage));
+	}
+
+	public static HttpResponse conflict(String errorMessage) {
+		return new HttpResponse(HttpStatus.CONFLICT, new HttpHeader(), new HttpResponseBody(errorMessage));
+	}
+
+	public static HttpResponse unsupportedMediaType(String errorMessage) {
+		return new HttpResponse(HttpStatus.UNSUPPORTED_MEDIA_TYPE, new HttpHeader(),
+			new HttpResponseBody(errorMessage));
+	}
+
+	public static HttpResponse internalServerError(String errorMessage) {
+		return new HttpResponse(HttpStatus.INTERNAL_SERVER_ERROR, new HttpHeader(), new HttpResponseBody(errorMessage));
+	}
+
+	public static HttpResponse staticResource(HttpResponseBody httpResponseBody, MIME mime) {
+		Map<String, String> headers = Map.of(HEADER_CONTENT_TYPE, mime.contentType, HEADER_CONTENT_LENGTH,
+			Integer.toString(httpResponseBody.getLength()));
+		return new HttpResponse(HttpStatus.OK, new HttpHeader(headers), httpResponseBody);
+	}
+
+	public byte[] makeByteMessage() {
+		StringBuilder header = new StringBuilder(
+			"HTTP/1.1 " + httpStatus.getCode() + " " + httpStatus.getDescription() + " \r\n");
```

Comment:
httpStatus 객체에게 메시지를 보내서 `"HTTP/1.1 " + httpStatus.getCode() + " " + httpStatus.getDescription() + " \r\n"` 의 응답을 만들어달라고 할 수 있지 않을까요? 🙂

```suggestion
		StringBuilder header = new StringBuilder(httpStatus.xxx());
```

Reply:
객체들에게 문자열 생성을 맡기고 밖에서 합치게 수정했습니다!